compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/test/tile_and_fuse_using_scf_forall.mlir

```diff
@@ -32,7 +32,25 @@ func.func @conv_2d_nhwc_hwcf(%arg0: tensor<2x14x14x32xbf16>, %arg1: tensor<3x3x3
 // TILE-LEVEL-0-SAME:   {
 //      TILE-LEVEL-0:       linalg.fill
 //      TILE-LEVEL-0:       linalg.conv_2d_nhwc_hwcf
-//      TILE-LEVEL-0:   }
+//      TILE-LEVEL-0:   }  {mapping = [#gpu.block<y>, #gpu.block<x>, #gpu.block<z>]}
```

Comment:
Yeah, so the logic is actually identical for threads and blocks. What's happening in this example is that the tiling sizes at thread and block levels are different:

For blocks:    `[0, 4, 4, 4, 0, 0, 0]`
For threads:  `[1, 1, 4, 4, 0, 0, 0]`

The logic works implemented in this PR works as follows:

First, assign attributes to dimensions with tile size greater than 1. For threads, that is dimensions 2 and 3.  
Second, assign attributes to dimensions with tile size equal to 1. For threads, that is dimensions 0 and 1. 

The attributes assigned in the order `y` then `x` then `z` then `linear_dim_0, linear_dim_1` etc.

For `[1, 1, 4, 4, 0, 0, 0]`, after step 1 the assigned dimensions are 
`[none, none, y, x, none, none, none]`
and then after step 2 the assigned dimensions are 
`[z, linear_dim_0, y, x, none, none, none]`. 

And that is why at the thread level we end up with [z, linear_dim_0, y, x].

Reply:
Thanks for the explanation. Yeah, I can follow the steps to set up mapping attributes. It's just not typical if we have different attributes for the same dim in block and thread mapping. I don't have a good solution to solve this other than hardcoded the dimensions. On the other hand, since the block attributes are not used anyway, maybe we could remove these block attributes?

Reply:
I just tried removing block dimension tiling, but for small matmul examples where the block tile sizes are all 1, there's a problem: scf.forall without tiling attributes get canonicalized away when the iterations space is size 1. i.e. the block-level scf.forall gets removed completely. Ideally we'd be able to work without this outer scf.forall, but currently the passes aren't set up to handle this, I guess. So removing block dimensions for matmuls isn't something we can immediately do. 

I could remove them for convolution, but we might have the same issue when we have small convolutions. 

So yeah, not sure. Maybe, for now, it's ok to keep the block dimensions as they are for convolution?

Reply:
I see... We should find a way to solve the scf.forall canonicalization problem. Could you add a TODO comment for the block mapping attributes?

Reply:
Ok, I'll add a comment. Thanks for accepting :)